code/src/containers/forms/product-form/hooks/use-create-product/useCreateProduct.ts

```diff
@@ -0,0 +1,34 @@
+import { useNavigate } from "react-router-dom";
+
+import routes from "@/constants/routes";
+import useSnackbar from "@/hooks/use-snackbar/useSnackbar";
+import { useCreateProductMutation } from "@/store/api/productsApi";
+import { CreateProductBody } from "@/types/product.types";
+
+const useCreateProduct = () => {
+  const [createProduct, requestState] = useCreateProductMutation();
+
+  const { openSnackbarWithTimeout } = useSnackbar();
+
+  const navigate = useNavigate();
+
+  const onSubmit = async (values: CreateProductBody) => {
+    try {
+      await createProduct(values).unwrap();
+      openSnackbarWithTimeout({
+        variant: "success",
+        messageTranslationKey: "productForm.creation.success"
+      });
+      navigate(routes.dashboard.products.path);
```

Comment:
I think it is better to redirect to page of single product in this case

Reply:
I am working on separate product details page for manager right now, I think I will change URL in that PR, what do you say?